src/compiler/checker.ts

```diff
@@ -21682,18 +21682,45 @@ namespace ts {
 
         function checkUnusedModuleMembers(node: ModuleDeclaration | SourceFile): void {
             if (compilerOptions.noUnusedLocals && !(node.flags & NodeFlags.Ambient)) {
+                const unusedImports: ImportedDeclaration[] = [];
                 node.locals.forEach(local => {
-                    if (!local.isReferenced && !local.exportSymbol) {
-                        for (const declaration of local.declarations) {
-                            if (!isAmbientModule(declaration)) {
-                                errorUnusedLocal(declaration, symbolName(local));
-                            }
+                    if (local.isReferenced || local.exportSymbol) return;
+                    for (const declaration of local.declarations) {
+                        if (isAmbientModule(declaration)) continue;
+                        if (isImportedDeclaration(declaration)) {
+                            unusedImports.push(declaration);

```

Comment:
Personally, I find the forEach code below a little hard to follow.  Another way to express this might be to build a multi-map from import clauses to the unused declarations they contain.  Then, for each import clause in the map, you could either report a single diagnostic or a list of diagnostics according to whether or not the import clause contained a declaration not in the corresponding unused list.  I don't feel strong about this.

Reply:
Thanks, helped me remove the `count` variable.